source/neuropod/bindings/java/src/main/native/com_uber_neuropod_NeuropodTensorAllocator.cc

```diff
@@ -0,0 +1,94 @@
+/* Copyright (c) 2020 UATC, LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "com_uber_neuropod_NeuropodTensorAllocator.h"
+
+#include "jclass_register.h"
+#include "neuropod/neuropod.hh"
+#include "utils.h"
+
+#include <jni.h>
+
+const int FLOAT  = 0;
+const int DOUBLE = 1;
+const int INT    = 2;
+const int LONG   = 3;
+
+using namespace neuropod::jni;
+
+JNIEXPORT void JNICALL Java_com_uber_neuropod_NeuropodTensorAllocator_nativeDelete(JNIEnv *env, jobject, jlong handle)
+{
+    try
+    {
+        auto allocator = reinterpret_cast<std::shared_ptr<neuropod::NeuropodTensorAllocator> *>(handle);
+        (*allocator).reset();
+        delete allocator;
+    }
+    catch (const std::exception &e)
+    {
+        neuropod::jni::throwJavaException(env, e.what());
+    }
+}
+
+JNIEXPORT jlong JNICALL Java_com_uber_neuropod_NeuropodTensorAllocator_nativeAllocate(
+    JNIEnv *env, jclass, jlongArray dims, jint typeNumber, jobject buffer, jlong handle)
```

Comment:
I don't follow about typeNumber, do we expect that user specifies it or this is deduced somehow? I don't see if this was demonstrated in Unit Test, please clarify.

Reply:
The typeNumber is the value we get when converting Java's enum to numbers. In Java side we have the following:     FLOAT_TENSOR(0),DOUBLE_TENSOR(1),INT32_TENSOR(2),INT64_TENSOR(3),STRING_TENSOR(4)
When call this function in the Java side, we use: 
nativeAllocate(dims, tensorType.getValue(), tensor.buffer, super.getNativeHandle()))
tensorType.getValue() convert a Java enum to int